src/main/java/de/bayern/gdi/gui/Controller.java

```diff
@@ -220,13 +225,16 @@ protected Integer call() throws Exception {
                     Platform.runLater(() -> {
                         progressSearch.setVisible(true);
                     });
-                    List<ServiceModel> catalog = dataBean.getCatalogService()
-                            .getServicesByFilter(currentText);
-                    for (ServiceModel entry: catalog) {
-                        dataBean.addCatalogServiceToList(entry);
-                        Platform.runLater(() -> {
-                            subentries.add(entry);
-                        });
+                    if (catalogReachable) {
+                        List<ServiceModel> catalog =
+                                dataBean.getCatalogService()
+                                .getServicesByFilter(currentText);
+                        for (ServiceModel entry : catalog) {
+                            dataBean.addCatalogServiceToList(entry);
+                            Platform.runLater(() -> {
+                                subentries.add(entry);
+                            });
```

Comment:
Mhm .. Wouldn't it be better to pull the `Platform.runLater` stuff out of the loop and do only one
```
Platform.runLater(() -> {
    subEntries.addAll(catalog);
}
```
afterwards, to reduce the overhead of passing the infos to the UI thread!?